htdocs/fourn/class/api_supplier_invoices.class.php

```diff
@@ -380,15 +380,15 @@ public function getPayments($id)
 			throw new RestException(400, 'Invoice ID is mandatory');
 		}
 
-		if (!DolibarrApi::_checkAccessToResource('fournisseur', $this->invoice->id, 'facture_fourn', 'facture')) {
-			throw new RestException(401, 'Access not allowed for login '.DolibarrApiAccess::$user->login);
-		}
-
 		$result = $this->invoice->fetch($id);
 		if (!$result) {
 			throw new RestException(404, 'Invoice not found');
 		}
 
+		if (!DolibarrApi::_checkAccessToResource('fournisseur', $this->invoice->id, 'facture_fourn', 'facture')) {
```

Comment:
This generate a security pb. A user is able to know if an invoice exists or not without having permission on it.
I think a better fix is to replace 
```
if (!DolibarrApi::_checkAccessToResource('fournisseur', $this->invoice->id, 'facture_fourn', 'facture')) {
			throw new RestException(401, 'Access not allowed for login '.DolibarrApiAccess::$user->login);
		}

```with 
```
if (!DolibarrApi::_checkAccessToResource('fournisseur', $id, 'facture_fourn', 'facture')) {
			throw new RestException(401, 'Access not allowed for login '.DolibarrApiAccess::$user->login);
		}
```

We keep to show the error "Access not allowed" before the "Invoice not found"

Reply:
I think so. Let's first fix the api to make it working...

Reply:
Thank you @eldy. Is thing going to be ported to 17.0/18.0 and develop as well?